### PR TITLE
Fix crash when serializing empty delegates in C#

### LINKED ISSUE #67212: Godot crashes with NullReferenceException in C# after rebuild
### Godot version

4.0.dev (17c62a692)

### System information

Windows 10, Vulkan, AMD Radeon RX 6700 XT

### Issue description

Hey there I get this exception method if I define a signal, create a  node using the script and after that rebuild the project.

I get this exception message when attaching to the Godot Process:
`System.NullReferenceException: Object reference not set to an instance of an object.
   at Godot.DelegateUtils.TrySerializeSingleDelegate(Delegate delegate, Byte[]& buffer) in C:\Users\alexa\source\git\contributions\Godot\Godot\modules\mono\glue\GodotSharp\GodotSharp\Core\DelegateUtils.cs:line 113
   at Godot.DelegateUtils.TrySerializeDelegate(Delegate delegate, Array serializedData) in C:\Users\alexa\source\git\contributions\Godot\Godot\modules\mono\glue\GodotSharp\GodotSharp\Core\DelegateUtils.cs:line 100
   at Godot.Bridge.GodotSerializationInfo.AddSignalEventDelegate(StringName name, Delegate eventDelegate) in C:\Users\alexa\source\git\contributions\Godot\Godot\modules\mono\glue\GodotSharp\GodotSharp\Core\Bridge\GodotSerializationInfo.cs:line 47
   at ProtoBots.Components.Templates.Template.SaveGodotObjectData(GodotSerializationInfo info) in C:\Users\alexa\source\git\godot\ProtoBots\Godot.SourceGenerators\Godot.SourceGenerators.ScriptSerializationGenerator\ProtoBots.Components.Templates.Template_ScriptSerialization_Generated.cs:line 14
   at Godot.Bridge.CSharpInstanceBridge.SerializeState(IntPtr godotObjectGCHandle, godot_dictionary* propertiesState, godot_dictionary* signalEventsState) in C:\Users\alexa\source\git\contributions\Godot\Godot\modules\mono\glue\GodotSharp\GodotSharp\Core\Bridge\CSharpInstanceBridge.cs:line 212`
   
   The issue is in line 113 where it access Target from a delegate value which is null.
`   object? target = @delegate.Target;`
   
   I guess this should be like this instead
`   object? target = @delegate?.Target;`
   

### Steps to reproduce

Define a signal in a C# Script. Create a scene where a node with this script is used.
Save the scene.

If you now rebuild your C# project using Rebuild Solution your editor should crash.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by dotlogix**
Resolves #67212

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs [hunk 1]
```diff
@@ -76,6 +76,9 @@ private enum TargetKind : uint
 
         internal static bool TrySerializeDelegate(Delegate @delegate, Collections.Array serializedData)
         {
+			if(@delegate is null)
```
Comment (akien-mga) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs:79]: Style issue (please see surrounding code and use the same style).

Comment (dotlogix) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs:79]: Line 94 and 97 share the same style without braces, but maybe this is better :)

Comment (neikeq) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs:79]: I think Akien meant:

```suggestion
			if (@delegate is null)
```

**Comment by akien-mga** [comment]
The rebase went wrong, you seem to have squashed your commits together with a previously merged commit. You might be able to fix it by resetting Tina previous version using `git reflog`, and then rebase on upstream master.

See PR workflow for tips: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

**Comment by dotlogix** [comment]
Ok finally I got it :D sorry @akien-mga for the messed up history :D
Not that familiar with the rebasing approach yet.

**Review by neikeq — APPROVED** [review]

File: modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs [hunk 2]
```diff
@@ -76,6 +76,11 @@ private enum TargetKind : uint
 
         internal static bool TrySerializeDelegate(Delegate @delegate, Collections.Array serializedData)
         {
+			if (@delegate is null)
+			{
```
Comment (akien-mga) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs:80]: There's trailing whitespace on this line that should be removed.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs [hunk 3]
```diff
@@ -76,6 +76,11 @@ private enum TargetKind : uint
...
+				return false;
+			}
```
Comment (akien-mga) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs:82]: Sorry for only noticing now... but I think C# code should be indented with 4 spaces and not tabs.

See `modules/mono/.editorconfig`, there should be a way for your IDE to pick it up and enforce this style properly.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**